source/WithMutex.h

```diff
@@ -0,0 +1,42 @@
+/* WithMutex.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef WITH_MUTEX_H_
+#define WITH_MUTEX_H_
+
+#include <mutex>
+
+
+
+// Helper class to add a mutex to Body without declaring the full copy constructor.
+// These operations are not thread safe. Do not copy the object in a concurrent context,
+// or with a locked mutex.
+// The use of the mutex must be enforced externally.
+class WithMutex
+{
+public:
+	WithMutex() = default;
+	WithMutex(const WithMutex &other);
+	WithMutex &operator=(const WithMutex &other);
```

Comment:
these copy constructors take one in but do not do anything with it, why not just delete them?

Reply:
Body needs to be copyable, but mutexes can't be copied.

Reply:
ok